app/scripts/lib/tx-utils.js

```diff
@@ -107,20 +83,17 @@ module.exports = class txProviderUtils {
   }
 
   publishTransaction (rawTx) {
-    return new Promise((resolve, reject) => {
-      this.query.sendRawTransaction(rawTx, (err, ress) => {
-        if (err) reject(err)
-        else resolve(ress)
-      })
-    })
+    return this.query.sendRawTransaction(rawTx)
   }
 
-  validateTxParams (txParams, cb) {
-    if (('value' in txParams) && txParams.value.indexOf('-') === 0) {
-      cb(new Error(`Invalid transaction value of ${txParams.value} not a positive number.`))
-    } else {
-      cb()
-    }
+  validateTxParams (txParams) {
```

Comment:
prefer `async` keyword instead of manually returning a promise
`reject(err)` -> `throw err`
`resolve(val)` -> `return val`